src/libOpenImageIO/imageinput.cpp

```diff
@@ -1120,11 +1117,11 @@ std::string
 ImageInput::geterror(bool clear) const
 {
     std::string e;
-    std::string* errptr = m_impl->m_errormessage.get();
-    if (errptr) {
-        e = *errptr;
+    auto iter = input_error_messages.find(this);
+    if (iter != input_error_messages.end()) {
+        e = iter.value();
         if (clear)
-            errptr->clear();
+            iter.value().clear();
```

Comment:
If this was `input_error_messages.erase(iter)`, instead of `clear()`, then any time an error was retrieved/clear by the right thread, the entry in the map would fully be cleaned up, instead of "leaking", even though the thing it leaks is usually an empty string.